sessionctx/variable/tidb_vars.go

```diff
@@ -229,4 +233,5 @@ var (
 	maxDDLReorgWorkerCount int32 = 128
 	// DDLSlowOprThreshold is the threshold for ddl slow operations, uint is millisecond.
 	DDLSlowOprThreshold uint32 = 300
+	ServerHostname, _          = os.Hostname()
```

Comment:
What if we later change the hostname?

Reply:
I checked the behavior of mysql 5.7 just now: if you change the hostname, MySQL will not pick it up.  Running `FLUSH HOSTS` does not clear the cache either.

I suspect this is either an accident, or relates to the MySQL team is trying to make sure `show global variables` does as little work as possible (performance risks from zealous monitoring tools).

Reply:
That is to say that this is MySQL compatible, even if by accident :-)